deposit_contract/contracts/validator_registration.v.py

```diff
@@ -65,22 +65,20 @@ def get_deposit_count() -> bytes[8]:
 @public
 def deposit(pubkey: bytes[PUBKEY_LENGTH],
             withdrawal_credentials: bytes[WITHDRAWAL_CREDENTIALS_LENGTH],
-            signature: bytes[SIGNATURE_LENGTH]):
+            signature: bytes[SIGNATURE_LENGTH],
+            deposit_data_root: bytes32):
     # Avoid overflowing the Merkle tree (and prevent edge case in computing `self.branch`)
     assert self.deposit_count < MAX_DEPOSIT_COUNT
 
-    # Validate deposit data
+    # Check deposit amount
     deposit_amount: uint256 = msg.value / as_wei_value(1, "gwei")
     assert deposit_amount >= MIN_DEPOSIT_AMOUNT
-    assert len(pubkey) == PUBKEY_LENGTH
```

Comment:
@daejunpark do you see any issue with removing the length checks here now that we have the added checksum via `deposit_data_root`

Reply:
As a side note, I think "checksum" is a bit misleading. The `deposit_data_root` check cryptographically strong :)

Reply:
fair! check-value :)
Still doesn't prevent _all_ error, but makes it way less likely to shoot yourself in the foot

Reply:
Think it might technically still count as a checksum -- https://en.wikipedia.org/wiki/Checksum

> a good checksum algorithm will usually output a significantly different value, even for small changes made to the input. This is especially true of cryptographic hash functions...

Reply:
Hmm, I would need to re-run the formal verification spec with this change to properly answer your question. So, the main question is: would the checksum be enough even when the argument arrays are of length smaller than the expected? (Note that the case of the lengths being lager than the expected will be detected and rejected in the beginning by the compiler-generated decoding process.)

Even if this change is logically OK (i.e., the formal verification spec is proved), I have a concern in case that all the argument arrays are empty. Since some of cryptographic functions (e.g., ecrecover) returns zero or the empty byte array in certain failure cases, it may be possible that some clients just use the empty byte array for computing the checksum and pass them to the deposit function. But one may argue that they must run `bls_verify` on the checksum, so I'm not sure how critical my concern is.

In any case, I'll come back to you once I re-run the formal verification spec with this.

Reply:
Vyper version 0.1.0b12 has been released and I have recompiled the contract in this PR to that version. I think this PR is ready for review :)